package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,703 @@
+        if self._has['force']:
+            ts.forces = self._get_frame_dataset('force')
+
+        if self.convert_units:
+            self._convert_units()
+
+        return ts
+
+    def _copy_to_data(self):
+        """assigns values to keys in data dictionary"""
+
+        if 'observables' in self._file:
+            for key in self._file['observables'].keys():
+                self.ts.data[key] = self._file['observables'][key][
+                                    'value'][self._frame]
+
+        # pulls 'time' out of first available parent group
+        for name, value in self._has.items():
+            if value:
+                if 'time' in self._particle_group[name]:
+                    self.ts.data['time'] = self._particle_group[name][
+                                           'time'][self._frame]
+                    break
+
+    def _get_frame_dataset(self, dataset):
+        """retrieves dataset array at current frame"""
+
+        frame_dataset = self._particle_group[
+                        dataset]['value'][self._frame, :]
+        n_atoms_now = frame_dataset.shape[0]
+        if n_atoms_now != self.n_atoms:
+            raise ValueError("Frame {} has {} atoms but the initial frame"
+                             " has {} atoms. MDAnalysis is unable to deal"
+                             " with variable topology!"
+                             "".format(self._frame,
+                                       n_atoms_now,
+                                       self.n_atoms))
+        return frame_dataset
+
+    def _convert_units(self):
+        """converts time, position, velocity, and force values if they
+        are not given in MDAnalysis standard units
+
+        See https://userguide.mdanalysis.org/1.0.0/units.html
+        """
+
+        self.ts.time = self.convert_time_from_native(self.ts.time)
+
+        if 'edges' in self._particle_group['box']:
+            self.convert_pos_from_native(self.ts.dimensions[:3])
+
+        if self._has['position']:
+            self.convert_pos_from_native(self.ts.positions)
+
+        if self._has['velocity']:
+            self.convert_velocities_from_native(self.ts.velocities)
+
+        if self._has['force']:
+            self.convert_forces_from_native(self.ts.forces)
+
+    def _read_next_timestep(self):
+        """read next frame in trajectory"""
+        return self._read_frame(self._frame + 1)
+
+    def close(self):
+        """close reader if ``driver="mpio"`` is not used to open file
+
+        Note
+        ----
+
+        If the `driver` and `comm` arguments were used to open the
+        hdf5 file (specifically, ``driver="mpio"``), then this method
+        does *not* close the file like most readers because
+        the information about the MPI communicator would be lost; instead
+        it leaves the file open by doing nothing.
+
+        """
+        if self._driver != 'mpio':
+            self._file.close()
```

Comment:
`close()` must close the file, that's just how all users will understand what `close()` means. we will also have problems with deleting objects (e.g., ``Reader.__del__()`` class close).

Please change so that it makes no exceptions.

What problem did this hack solve?

Reply:
When `_reopen` wasn't closing the file in an earlier commit, `close` was causing some base tests to fail